crates/wasmtime/src/func.rs

```diff
@@ -194,9 +194,17 @@ macro_rules! getters {
                     >(export.address);
                     let mut ret = None;
                     $(let $args = $args.into_abi();)*
-                    catch_traps(export.vmctx, &instance.store, || {
-                        ret = Some(fnptr(export.vmctx, ptr::null_mut(), $($args,)*));
-                    })?;
+
+                    {
+                        let canary = 0;
+                        let _auto_reset = instance
```

Comment:
It probably doesn't matter in practice, but that would mean the canary is taken from a stack frame that has since been popped

/me shrugs

I guess I could switch the API to `table.with_stack_canary(|| { ... })` instead of an RAII thing? Would be harder to misuse this way too. I think I'll do that.

Reply:
Either way's fine by me, I was mostly just hoping that we can encapsulate everything necessary to enter wasm in one function in wasmtime, which currently is `catch_traps`. I think the `with_stack_canary` call can be embedded in there too? (we can perhaps rename `catch_traps` to `invoke_wasm` or something like that)

Reply:
Ah I misunderstood what you were asking for in the original comment. This is done now.